src/nvim/path.c

```diff
@@ -2186,9 +2186,16 @@ static int path_get_absolute_path(const char_u *fname, char_u *buf,
 
 /// Check if the given file is absolute.
 ///
-/// This just checks if the file name starts with '/' or '~'.
 /// @return `TRUE` if "fname" is absolute.
 int path_is_absolute_path(const char_u *fname)
 {
+#ifdef WIN32
+  // A name like "d:/foo" and "//server/share" is absolute
+  return (isalpha(fname[0]) && fname[1] == ':'
+          && vim_ispathsep_nocolon(fname[2]))
+           || (vim_ispathsep_nocolon(fname[0]) && fname[0] == fname[1]);
```

Comment:
Also can you wrap the whole expression in `()` so that opening is right after `return` and closing is right before `;`? Current alignment is not correct, `||` must be less aligned then `&&` on the previous line and using additional set of parens makes linter leave you with the only correct choice.